web/src/scripts/webflow/fundingSync.mjs

```diff
@@ -279,15 +286,20 @@ const updateFundings = async () => {
       return await modifyItem(item.id, fundingCollectionId, funding);
     } catch (error) {
       console.error(error);
+      if (error.response.status === 429){
+        console.debug("Catching rate limit error hit")
+        await wait(65000);
+        return await modifyItem(item.id, fundingCollectionId, funding);
+      }
```

Comment:
I ended up handling the errors inline everywhere. 

I realized that because we want to retry and recall this function we woudl still need to pass the function (in this case modifyItem and all its params)

so we could have done something like this 

```
async function handleRateLimitError(func, ...params) {
   if (error.response && error.response.status === 429) {
      console.debug("Catching rate limit error hit");
      await wait(65000);
      return await func(...params);
    }
}


try {
  return await modify( item.id, fundingCollectionId, funding);
} catch (error) {
  console.debug({ status: error.response.status, type: typeof error.response.status });
  await handleRateLimitError(modify, item.id, fundingCollectionid, funding);
  throw error;
}
```

Reply:
I think this would work but something about it kinda bothers me. I can do it if people feel strongly.

Reply:
Given how many times you're repeating roughly the same logic, this would be a good opportunity to make a function. It's not a common pattern for us, but I don't think there's anything wrong with it. I'm in the camp of make-it-a-function.

Reply:
At the very least, you should make `65000` a global const.

Reply:
[boulder] Yes, please factor out. Also, you'll need to pass the error, or at least the response, to `handleRateLimitError()`. Also, as discussed, I'd prefer if the `throw error` occurred in the `handleRateLimitError()` instead, on the logic branch where `error.response.status` was not 429.